atlasdb-client/src/test/java/com/palantir/atlasdb/util/AtlasDbMetricsTest.java

```diff
@@ -91,6 +122,77 @@ public void instrumentWithCustomNameAll() {
         assertMethodInstrumented(MetricRegistry.name(CUSTOM_METRIC_NAME, PING_NOT_TIMED_METHOD), service::pingNotTimed);
     }
 
+    @Test
+    public void instrumentTaggedAsyncFunction() throws InterruptedException, ExecutionException {
+        AsyncTestService asyncTestService = AtlasDbMetrics.instrumentWithTaggedMetrics(taggedMetrics,
+                AsyncTestService.class, this.asyncTestService);
+
+        String asyncPingMetricName = MetricRegistry.name(AsyncTestService.class, "asyncPing");
+        assertTimerNotRegistered(asyncPingMetricName);
+
+        List<ListenableFuture<String>> futures = IntStream.range(0, 10)
+                .mapToObj($ -> asyncTestService.asyncPing())
+                .collect(Collectors.toList());
+
+        Instant now = Instant.now();
+        MetricName metricName = MetricName.builder().safeName(asyncPingMetricName).build();
+        ListenableFuture<Boolean> done = Futures.whenAllSucceed(futures).call(() -> {
+            // have to do it this because we can't edit the future we get back and it's only a callback as opposed to a
+            // transformed future
+            Awaitility.await()
+                    .atMost(ASYNC_DURATION_TTL.toMillis(), TimeUnit.MILLISECONDS)
+                    .until(() -> taggedMetrics.timer(metricName).getSnapshot().size() > 0);
+            return true;
+        }, MoreExecutors.directExecutor());
+
+        assertThat(done.get()).isEqualTo(true);
+
+        assertThat(done.get()).isEqualTo(true);
```

Comment:
Should there be two of the exact same statement here? If so, please may you leave a comment as to why?

nit: maybe also group them together w.r.t spacing?

Reply:
deleted, can't remember, was likely an accident